lib/kcerts/ssh.go

```diff
@@ -0,0 +1,125 @@
+package kcerts
+
+import (
+	"bufio"
+	"bytes"
+	"crypto/rand"
+	"crypto/rsa"
+	"fmt"
+	"github.com/mitchellh/go-homedir"
+	"golang.org/x/crypto/ssh"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"time"
+)
+
+const CAPrefix = "@cert-authority"
+const SSHDir = ".ssh"
+const KnownHosts = "known_hosts"
+
+// AddSSHCAToClient adds a public key to the $HOME/.ssh/known_hosts in the ssh-cert x509.1 format.
+// For each entry, it adds an additional line and does not concatenate
+func AddSSHCAToClient(publicKey ssh.PublicKey, hosts []string) error {
+	hDir, err := homedir.Dir()
+	if err != nil {
+		return err
+	}
+	sshDir := filepath.Join(hDir, SSHDir)
+	if _, err := os.Stat(sshDir); os.IsNotExist(err) {
```

Comment:
see larger explanation below - but this block, line 31 to 37, is unnecessary. If the directory does not exist, your attempt to read the `known_hosts` file will probably return "No such file or directory", which is perfectly clear.